internal/js/modules/k6/browser/common/page.go

```diff
@@ -514,6 +517,35 @@ func (p *Page) onRequest(request *Request) {
 	}
 }
 
+// onResponse will call the handlers for the page.on('response') event.
+func (p *Page) onResponse(resp *Response) {
+	p.logger.Debugf("Page:onResponse", "sid:%v url:%v", p.sessionID(), resp.URL())
+
+	if !hasPageOnHandler(p, EventPageResponseCalled) {
+		return
+	}
+
+	p.eventHandlersMu.RLock()
+	defer p.eventHandlersMu.RUnlock()
+	for _, h := range p.eventHandlers[EventPageResponseCalled] {
+		err := func() error {
+			// Handlers can register other handlers, so we need to
+			// unlock the mutex before calling the next handler.
+			p.eventHandlersMu.RUnlock()
+			defer p.eventHandlersMu.RLock()
+
+			// Call and wait for the handler to complete.
+			return h(PageOnEvent{
+				Response: resp,
+			})
+		}()
+		if err != nil {
+			p.logger.Warnf("onResponse", "handler returned an error: %v", err)
+			return
+		}
+	}
+}
```

Comment:
We start to duplicate this logic in `Page.urlTagName`, `Page.onRequest`, and `Page.onConsoleAPICalled` (although this one differs a little). What do you think about introducing a helper function in #4290?

Reply:
Yeah, i had a feeling you were going to point that out 😄 Sure, i think that should be done in a separate refactor PR 👍

Reply:
I refactored the test based on comments made in https://github.com/grafana/k6/pull/4290.